helm/values-prod.yaml

```diff
@@ -128,6 +128,14 @@ microservice-chart:
     limits:
       memory: "512Mi"
       cpu: "500m"
+  deployment:
+    create: true
+    replicas: 3
+  strategy:
+    type: "RollingUpdate"
+    rollingUpdate:
+      maxSurge: "100%"
```

Comment:
Maybe you can set `maxSurge` to 0.
`maxSurge`  "_specifies the maximum number of Pods that can be created over the desired number of Pods._"

https://kubernetes.io/docs/concepts/workloads/controllers/deployment/

If you set `"100%"` and have a number of pods set to `n`, you could have `n*2` starting pods.

Reply:
To be completely honest, for this particular microservice this condition will never be dangerous, since the receivers are stopped before the deploy.

Reply:
Yes, this was set on purpose based on the fact that event receivers will be stopped before starting the deployment rollout phase, so in this particular case this configuration minimize the deployment time